cabot/cabotapp/tests/test_schedules.py

```diff
@@ -138,3 +143,163 @@ def test_calendar_long_name(self):
+
+If you don't want to deal with this right now, you can temporarily silence these alerts from the page linked above."""\
+            .format(settings.WWW_SCHEME, settings.WWW_HTTP_HOST, self.schedule.pk)
+
+        self.assertTrue(fake_send_mail.called)
+        fake_send_mail.assert_called_with(('Principal', message, [u'dolores@affirm.com']))
+
+    @patch('cabot.cabotapp.models.requests.get', fake_calendar)
+    @patch('cabot.cabotapp.models.send_schedule_problems_email.apply_async')
+    @patch('cabot.cabotapp.models.timezone.now')
+    def test_validate_schedule_problems_fixed(self, fake_now, fake_send_email):
+        """Test that schedule.problems gets set then cleared after we fix the problems"""
+
+        fake_now.return_value = datetime(2018, 8, 15, 0, 3, 54, 598552, tzinfo=timezone.utc)
+
+        # start with 2 problems
+        self.schedule.ical_url = 'calendar_with_gaps.ics'  # bad calendar
+        self.schedule.fallback_officer = None  # no fallback officer
+        self.schedule.save()
+        # update_shifts(self.schedule) and self.schedule.update_problems() automatically called by post_save handler
+        self.schedule = Schedule.objects.get(pk=self.schedule.pk)  # refresh from DB
+
+        # check that schedule.problems gets set correctly
+        problems_str = """\
+The schedule has no fallback officer.
+
+There are gaps in the schedule:
+* 2018-08-17 19:00:00+00:00 to 2018-08-18 19:00:00+00:00
+* 2018-08-19 19:00:00+00:00 to 2018-08-20 19:00:00+00:00\
+"""
+        self.assertEquals(self.schedule.problems, problems_str)
+
+        # change the calendar to a valid one (still no fallback officer)
+        self.schedule.ical_url = 'calendar_no_gaps.ics'
+        self.schedule.save()
+        # update_shifts(self.schedule) and self.schedule.update_problems() automatically called by post_save handler
+        self.schedule = Schedule.objects.get(pk=self.schedule.pk)  # refresh from DB
+
+        self.assertEquals(self.schedule.problems, "The schedule has no fallback officer.")
+
+        # fix the fallback officer (now there are no problems)
+        self.schedule.fallback_officer = User.objects.filter(email='dolores@affirm.com').first()
+        self.schedule.save()
+        # update_shifts(self.schedule) and self.schedule.update_problems() automatically called by post_save handler
+        self.schedule = Schedule.objects.get(pk=self.schedule.pk)  # refresh from DB
+
+        # verify that problems got cleared
+        self.assertIsNone(self.schedule.problems)
+        # no emails will ever send because we have no fallback officer for most of this test
+        self.assertFalse(fake_send_email.called)
+
+    @patch('cabot.cabotapp.models.requests.get', fake_calendar)
+    @patch('cabot.cabotapp.models.send_schedule_problems_email.apply_async')
+    @patch('cabot.cabotapp.models.timezone.now')
+    def test_validate_schedule_silence_until(self, fake_now, fake_send_email):
```

Comment:
I think it'd be good to split this out into 2 tests (one for UI and one for emails)...in general I think it's better for a test to check if something specific works so that if you see a failure you know exactly what's wrong. This is essentially testing 2 independent components (UI and email sending), so a failure could mean multiple things (and if the test fails on UI before you get to emails you don't know if emails are working or not, so you could fix a bug and have the test still fail).